pymeasure/experiment/results.py

```diff
@@ -129,16 +131,54 @@ def __init__(self, columns, delimiter=','):
         """
         super().__init__()
         self.columns = columns
+        self.units = self._parse_columns(columns)
         self.delimiter = delimiter
 
+    @staticmethod
+    def _parse_columns(columns):
+        """Parse the columns to get units in parenthesis."""
+        units_pattern = r"\((?P<units>[\w/\(\)\*\t]+)\)"
+        units = {}
+        for column in columns:
+            try:
+                match = re.search(units_pattern, column)
+            except TypeError:
+                match = None
+
+            if match:
+                m = match.groupdict()
+                if 'units' in m:
+                    units[column] = pint.Unit(m['units'])
+        return units
+
     def format(self, record):
         """Formats a record as csv.
 
         :param record: record to format.
         :type record: dict
         :return: a string
         """
-        return self.delimiter.join(f'{record[x]}' for x in self.columns)
+        line = []
+        for x in self.columns:
+            value = record.get(x, float("nan"))
+            units = self.units.get(x, u.dimensionless)
```

Comment:
careful, dimensionless (column `Mach number (1)`) and no unit (`device_mode`) are not the same. The former will have an entry in units (I suspect `u.dimensionless`), but here you also assign it to the latter. Maybe `None` is a better default?

Reply:
At the point of writing that code, I had "dimensionless" as a synonym of "I don't care", such that Mach number and device_mode both fit.
Let's talk about the implementation again, after having decided on the usage.

Reply:
Yeah, but you _should_ care about dimensionless things -- in the end the way you can think of it is that it has a unit of `1`. It still has to fit unit compatibility in operations, something which device_mode does not, so there is a difference, too.
>Let's talk about the implementation again, after having decided on the usage.

:+1:

Reply:
I adjusted the implementation (and tests) according to the table in the discussion #248